sunpy/visualization/visualization.py

```diff
@@ -1,27 +1,31 @@
 """
 This module provides plotting support in iPython.
 """
-from matplotlib import pyplot as plt
+from functools import wraps
 
-__all__ = ["toggle_pylab", "axis_labels_from_ctype"]
+import matplotlib.pyplot as plt
 
+__all__ = ['peek_show', "axis_labels_from_ctype"]
 
-def toggle_pylab(fn):
+
+def peek_show(func):
     """
-    A decorator to prevent functions from opening Matplotlib windows
-    unexpectedly when SunPy is run in interactive shells like iPython.
+    A decorator to place on ``peek()`` methods to show the figure.
 
-    Toggles the value of `matplotlib.pyplot.isinteractive`.
+    The ``peek()`` method should return the figure then this method will
+    attempt to show it in the correct way. This decorator will not return the
+    figure to the user.
     """
-    if plt.isinteractive():
-        def fn_itoggle(*args, **kwargs):
-            plt.ioff()
-            ret = fn(*args, **kwargs)
-            plt.ion()
-            return ret
-        return fn_itoggle
-    else:
-        return fn
+    @wraps(func)
+    def show_if_interactive(*args, **kwargs):
+        figure = func(*args, **kwargs)
+        # Show the figure if using an interactive Matplotlib backend
+        if mpl.get_backend() in mpl.rcsetup.interactive_bk:
```

Comment:
You can just get rid of this if check completely.

Reply:
Does `plt.show` just no-op if it's not an interactive backend? i.e. `inline`?

Reply:
Yeah, but there's no harm in calling it. In addition, I'm not sure if this if statement would work for the (very very niche) use case where someone is using a custom backend that doesn't ship with Matplotlib.